### PR TITLE
First set of changes for persistent repository.

### PR DESCRIPTION
**Pull request by jackgr**
Initial PR for persistence based on MongoDB. Includes all code changes and tests correctly against a localhost instance of MongoDB. Automated tests fail because MongoDB is not running in the test environment. Next PR will automate setup of MongoDB service in the target cluster when dm is installed, and unblock unit tests.

### CODE REVIEW

**Comment by jackgr** [comment]
@vaikas-google updated to skip the tests if mongodb is not running. Now ready to go. PTAL

**Comment by jtblin** [comment]
Could the persistence layer made pluggable by using an interface? I don't have anything against mongodb but it isn't in our current stack so I'd rather not have to care about it. Apologies if it isn't the right forum to discuss this, but I was curious about why we picked mongodb and not make it pluggable like following the `cloudproviders` pattern in kubernetes for example.

**Comment by jackgr** [comment]
@jtblin: it is already an interface. See repository/repository.go. There are currently two implementations, in-memory and persistent. The latter is MongoDB. Should be easy enough to add another. I suppose we could provide a repository plug-in API to make it a little cleaner. Can you open an issue for it?

**Comment by jtblin** [comment]
ok I see. Sorry about that, I just looked quickly at the diff and it was looking like we were just adding mongodb in `manager/deployments.go`. Yes a plug-in API would make it cleamer, I'll open an issue about that. Thanks for the explanation.

**Comment by jackgr** [comment]
Thanks.

Jack Greenfield | Google

On Fri, Feb 5, 2016 at 3:49 PM, Jerome Touffe-Blin <notifications@github.com

File: manager/repository/persistent/persistent.go [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+	"time"
+
+	"github.com/kubernetes/deployment-manager/common"
+	"github.com/kubernetes/deployment-manager/manager/repository"
+
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+type pDeployment struct {
```
Comment (vaikas-google) [inline manager/repository/persistent/persistent.go]: why not just deployment?

**Comment by jackgr** [comment]
@technosophos PTAL

File: manager/repository/transient/transient.go [hunk 2]
```diff
@@ -0,0 +1,325 @@
+/*
```
Comment (vaikas-google) [inline manager/repository/transient/transient.go]: Is this just a rename for the existent repository?

File: manager/repository/persistent/persistent.go (hunk 1 above)
Comment (jackgr) [inline manager/repository/persistent/persistent.go]: This is a persistent deployment. There's a transient deployment named tDeployment in transient.go. Yes, they're in different packages, so they could be the same, but given that we're defining types that differ significantly, I thought it better to give them different names.

File: manager/repository/transient/transient.go (hunk 2 above)
Comment (jackgr) [inline manager/repository/transient/transient.go]: Yes. We have both in service. Manager falls back the transient repository if the persistent one is not available.

**Comment by jackgr** [comment]
@vaikas-google Can I get an LGTM?

**Comment by vaikas-google** [comment]
LGTM

**Comment by jackgr** [comment]
Fixes #52.

**Pull request merged**